gap/Forms.gi

```diff
@@ -0,0 +1,90 @@
+# Note that <group> must have the attribute InvariantBilinearForm or
+# InvariantSesquilinearForm for this to work.
+InstallGlobalFunction("ChangeFixedSesquilinearForm",
+function(group, type, gramMatrix)
+    local gapForm, newForm, gapToCanonical, canonicalToNew, field;
+    if not type in ["S", "O", "U"] then
+        ErrorNoReturn("<type> must be one of 'S', 'U', 'O', but <type> = ",
+                      type);
+    fi;
+    field := DefaultFieldOfMatrixGroup(group);
+    if type = "S" or type = "O" then
+        gapForm := BilinearFormByMatrix(InvariantBilinearForm(group).matrix, 
+                                        field);
+        newForm := BilinearFormByMatrix(gramMatrix, field);
+    else
+        gapForm := HermitianFormByMatrix(InvariantSesquilinearForm(group).matrix,
+                                         field);
+        newForm := HermitianFormByMatrix(gramMatrix, field);
+    fi;
+    # the following if condition can only ever be fulfilled if <group> is an
+    # orthogonal group; there the case of even dimension is problematic since,
+    # in that case, there are two similarity classes of bilinear forms
+    if not WittIndex(gapForm) = WittIndex(newForm) then
+       ErrorNoReturn("The form preserved by <group> must be similar to the form ", 
+                     "described by the Gram matrix <gramMatrix>.");
+    fi;
+    gapToCanonical := BaseChangeHomomorphism(BaseChangeToCanonical(gapForm), 
+                                             field);
+    canonicalToNew := BaseChangeHomomorphism(BaseChangeToCanonical(newForm) ^ (-1), 
+                                             field);
+    return Group(canonicalToNew(gapToCanonical(GeneratorsOfGroup(group))));
+end);
+
+ConjugateModule := function(M, q)
+  return GModuleByMats(List(MTX.Generators(M), A -> ApplyFunctionToEntries(A, x -> x ^ q)), 
+                       MTX.Field(M));
+end;
+
+# Assuming that the group G acts absolutely irreducibly, try to find a unitary
+# form which is G-invariant or prove that no such form exists.
+InstallGlobalFunction("UnitaryForm",
```

Comment:
Have you considered using [`PreservedSesquilinearForms`](https://gap-packages.github.io/forms/doc/chap4.html#X84056A357E5447AF) from the GAP package `forms` instead?

Reply:
Yes and that works *reaaaaally* badly.

Reply:
Maybe there is something wrong with `PreservedSesquilinearForms` as such? - Namely, I am guessing if it worked correctly, it should (despite using randomised methods) recognise a non-trivial preserved unitary form at least once in 10 000 trials ...?

Reply:
Sure, a bug is possible and this would ideally be reported via the `forms` issue tracker.

At the same time, please document this kind of reason in a code comment. Like, 
> not using FOO because it is slow/bugggy/INSERT-REASON as the example XYZ illustrates

Reply:
Comment added :) Where can I find the issue tracker? As you point out yourself in the GitHub repo of `forms`, the link on the `forms` homepage is broken ...

Reply:
As usual it is part of the repo :-)

Reply:
See gap-packages/forms#22.